lms/djangoapps/mobile_api/utils.py

```diff
@@ -1,31 +1,92 @@
 """
-Tests for video outline API
+Common utility methods and decorators for Mobile APIs.
 """
 
 
+import functools
+from django.http import Http404
+
+from opaque_keys.edx.keys import CourseKey
 from courseware import access
-from student.roles import CourseBetaTesterRole
-from student import auth
+from courseware.courses import get_course_with_access
+from rest_framework import permissions
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
 
 
-def mobile_available_when_enrolled(course, user):
+def mobile_course_access(depth=0, verify_enrolled=True):
     """
-    Determines whether a user has access to a course in a mobile context.
-    Checks if the course is marked as mobile_available or the user has extra permissions
-    that gives them access anyway.
-    Does not check if the user is actually enrolled in the course
+    Method decorator for a mobile API endpoint that verifies the user has access to the course in a mobile context.
     """
+    def _decorator(func):
+        """Outer method decorator."""
+        @functools.wraps(func)
```

Comment:
Maybe this isn't a big deal, but I believe (and my python is not amazing so I'm not totally sure) that this will make a function that claims it takes an argument called ``course``, but actually takes an argument called ``course_id``?